test/OptionSettlementEngine.t.sol

```diff
@@ -519,10 +524,129 @@ contract OptionSettlementTest is Test, NFTreceiver {
         _assertClaimAmountExercised(claimIds[6], 0);
     }
 
+    function testWriteRecordFees() public {
```

Comment:
There are further tests we could write, re event emissions and multiple asset fee accounting, but wanted to open PR asap.